keyboards/dtd/ldmlKeyboard.xsd

```diff
@@ -0,0 +1,393 @@
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:attribute>
+    </xs:complexType>
+  </xs:element>
+  <!-- @ORDERED -->
+  <!-- @VALUE -->
+  <!-- @VALUE -->
+  <!-- @VALUE -->
+  <xs:element name="vkeyMaps">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="import"/>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="vkeyMap"/>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="special"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="vkeyMap">
+    <xs:complexType>
+      <xs:attribute name="from" use="required"/>
+      <xs:attribute name="to" use="required"/>
+    </xs:complexType>
+  </xs:element>
+  <!-- @MATCH:any -->
+  <!-- @MATCH:any -->
+  <!-- @VALUE -->
+  <xs:element name="displayMap">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="import"/>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="display"/>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="displayOptions"/>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="special"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="display">
+    <xs:complexType>
+      <xs:attribute name="to" use="required"/>
+      <xs:attribute name="display" use="required"/>
+    </xs:complexType>
+  </xs:element>
+  <!-- @MATCH:any -->
+  <!-- @MATCH:any -->
+  <!-- @VALUE -->
+  <xs:element name="displayOptions">
+    <xs:complexType>
+      <xs:attribute name="baseCharacter"/>
+    </xs:complexType>
+  </xs:element>
+  <!-- @MATCH:any -->
+  <!-- @VALUE -->
+  <xs:element name="keys">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="import"/>
+        <xs:choice minOccurs="0" maxOccurs="unbounded">
+          <xs:element ref="key"/>
+          <xs:element ref="flicks"/>
+        </xs:choice>
+        <xs:element minOccurs="0" maxOccurs="unbounded" ref="special"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="key">
+    <xs:complexType>
+      <xs:attribute name="id" use="required"/>
+      <xs:attribute name="flicks" type="xs:NMTOKEN"/>
+      <xs:attribute name="gap">
+        <xs:simpleType>
+          <xs:restriction base="xs:token">
+            <xs:enumeration value="true"/>
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="to"/>
+      <xs:attribute name="longPress"/>
+      <xs:attribute name="longPressDefault"/>
+      <xs:attribute name="multitap"/>
+      <xs:attribute name="switch" type="xs:NMTOKEN"/>
+      <xs:attribute name="transform">
+        <xs:simpleType>
+          <xs:restriction base="xs:token">
+            <xs:enumeration value="no"/>
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="width"/>
```

Comment:
For example, the numeric `@MATCH:range` argument could make this:

`name="width" type="xs:decimal" minInclusive="0.01" maxInclusive="100.0"`